Fix linting issues in gui_tools

For all of [these issues](https://github.com/feder-observatory/stellarphot/issues?q=is%3Aopen+is%3Aissue+label%3Alinting), first delete the appropriate line in `pyproject.toml` that excludes the subpackage from checks. For `gui_tools` you would delete [this line](https://github.com/feder-observatory/stellarphot/blob/main/pyproject.toml#L122):

```toml
"stellarphot/gui_tools/*" = ["F", "ARG"]
```

After that, run `ruff` either with `tox -e lint` or `ruff .`. The errors you get should be the ones -- fix them! If it is an unused function argument make sure to search the project for calls to that function to update the arguments there too. If it is an `import *` in a file then replace it with an explicit import of the functions actually used.

- [ ] stellarphot/gui_tools/comparison_functions.py:20:1: F403 `from stellarphot.differential_photometry import *` used; unable to detect undefined names
- [ ]stellarphot/gui_tools/comparison_functions.py:21:1: F403 `from stellarphot.photometry import *` used; unable to detect undefined names
- [ ] stellarphot/gui_tools/comparison_functions.py:120:12: ARG001 Unused function argument: `viewer`
- [ ] stellarphot/gui_tools/comparison_functions.py:120:27: ARG001 Unused function argument: `data_x`
- [ ] stellarphot/gui_tools/comparison_functions.py:120:35: ARG001 Unused function argument: `data_y`
- [ ] stellarphot/gui_tools/comparison_functions.py:127:9: F841 Local variable `pad` is assigned to but never used
- [ ] stellarphot/gui_tools/comparison_functions.py:128:9: F841 Local variable `x` is assigned to but never used
- [ ] stellarphot/gui_tools/comparison_functions.py:129:9: F841 Local variable `y` is assigned to but never used
- [ ] stellarphot/gui_tools/comparison_functions.py:368:25: ARG002 Unused method argument: `change`
- [ ] stellarphot/gui_tools/comparison_functions.py:389:39: ARG002 Unused method argument: `button`
- [ ] stellarphot/gui_tools/comparison_functions.py:414:38: ARG002 Unused method argument: `button`
- [ ] stellarphot/gui_tools/comparison_functions.py:566:31: ARG002 Unused method argument: `button`
- [ ] stellarphot/gui_tools/fits_opener.py:52:39: ARG005 Unused lambda argument: `x`
- [ ] stellarphot/gui_tools/photometry_widget_functions.py:83:35: ARG002 Unused method argument: `change`
- [ ] stellarphot/gui_tools/photometry_widget_functions.py:91:41: ARG002 Unused method argument: `change`
- [ ] stellarphot/gui_tools/seeing_profile_functions.py:476:9: F841 Local variable `layout` is assigned to but never used
- [ ] stellarphot/gui_tools/seeing_profile_functions.py:522:25: ARG002 Unused method argument: `file`
- [ ] stellarphot/gui_tools/seeing_profile_functions.py:545:40: ARG002 Unused method argument: `change`
- [ ] stellarphot/gui_tools/seeing_profile_functions.py:558:33: ARG002 Unused method argument: `button`
- [ ] stellarphot/gui_tools/seeing_profile_functions.py:587:33: ARG002 Unused method argument: `button`
- [ ] stellarphot/gui_tools/seeing_profile_functions.py:589:24: F821 Undefined name `ap_rad`
- [ ] stellarphot/gui_tools/seeing_profile_functions.py:589:33: F821 Undefined name `ap_rad`
- [ ] stellarphot/gui_tools/seeing_profile_functions.py:589:47: F821 Undefined name `ap_rad`
- [ ] stellarphot/gui_tools/seeing_profile_functions.py:623:24: ARG001 Unused function argument: `viewer`
- [ ] stellarphot/gui_tools/seeing_profile_functions.py:623:44: ARG001 Unused function argument: `datax`
- [ ] stellarphot/gui_tools/seeing_profile_functions.py:623:56: ARG001 Unused function argument: `datay`

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -122,7 +122,6 @@ select = [
 "docs/conf.py" = ["F405"]
 
 # Ignore pyflakes and ARGS issues in each subpackage until they are fixed
-"stellarphot/gui_tools/*" = ["F", "ARG"]
 "stellarphot/utils/*" = ["F", "ARG"]
 
 [tool.pytest.ini_options]

--- a/stellarphot/gui_tools/comparison_functions.py
+++ b/stellarphot/gui_tools/comparison_functions.py
@@ -125,9 +125,7 @@ def wrap(imagewidget, outputwidget):
             imagewidget.next_elim += 1
         except AttributeError:
             imagewidget.next_elim = 1
-        pad = 15
-        x = int(np.floor(event.data_x))
-        y = int(np.floor(event.data_y))
+
         ra, dec = i.wcs.wcs.all_pix2world(event.data_x, event.data_y, 0)
         out_skycoord = SkyCoord(ra=ra, dec=dec, unit=(u.degree, u.degree))
 

--- a/stellarphot/gui_tools/comparison_functions.py
+++ b/stellarphot/gui_tools/comparison_functions.py
@@ -17,8 +17,6 @@ except ImportError:
     from astrowidgets.ginga import ImageWidget
 
 from stellarphot import SourceListData
-from stellarphot.differential_photometry import *
-from stellarphot.photometry import *
 from stellarphot.gui_tools.seeing_profile_functions import set_keybindings
 from stellarphot.gui_tools.fits_opener import FitsOpener
 from stellarphot.io import TessSubmission, TOI, TessTargetFile

--- a/stellarphot/gui_tools/seeing_profile_functions.py
+++ b/stellarphot/gui_tools/seeing_profile_functions.py
@@ -409,7 +409,6 @@ class SeeingProfileWidget:
     def _update_plots(self):
         # DISPLAY THE SCALED PROFILE
         fig_size = (10, 5)
-        profile_size = 60
 
         rad_prof = self.rad_prof
         self.seeing_profile_plot.clear_output(wait=True)

--- a/stellarphot/gui_tools/seeing_profile_functions.py
+++ b/stellarphot/gui_tools/seeing_profile_functions.py
@@ -336,8 +336,8 @@ class SeeingProfileWidget:
 
     def _make_show_event(self):
         def show_event(
-            viewer, event=None, datax=None, datay=None, aperture=None
-        ):  # noqa: ARG001
+            viewer, event=None, datax=None, datay=None, aperture=None  # noqa: ARG001
+        ):
             """
             ginga callbacks require the function signature above.
             """

--- a/stellarphot/gui_tools/seeing_profile_functions.py
+++ b/stellarphot/gui_tools/seeing_profile_functions.py
@@ -227,15 +227,9 @@ class SeeingProfileWidget:
         self._set_observers()
         self.aperture_settings.description = ""
 
-    def load_fits(self, file):
+    def load_fits(self):
         """
         Load a FITS file into the image widget.
-
-        Parameters
-        ----------
-
-        file : str
-            Filename to open.
         """
         self.fits_file.load_in_image_widget(self.iw)
         self.object_name = self.fits_file.object
@@ -250,8 +244,9 @@ class SeeingProfileWidget:
             )
             self.exposure = np.nan
 
-    def _update_file(self, change):
-        self.load_fits(change.selected)
+    def _update_file(self, change):  # noqa: ARG002
+        # Widget callbacks need to accept a single argument, even if it is not used.
+        self.load_fits()
 
     def _construct_tess_sub(self):
         file = self.fits_file.path

--- a/stellarphot/gui_tools/tests/test_seeing_profile.py
+++ b/stellarphot/gui_tools/tests/test_seeing_profile.py
@@ -1,4 +1,3 @@
-from astropy.table import Table
 from astrowidgets import ImageWidget
 
 from stellarphot.gui_tools import seeing_profile_functions as spf

